modules/apps/document-library/document-library-video/src/main/java/com/liferay/document/library/video/internal/preview/DLVideoDLPreviewRendererProvider.java

```diff
@@ -63,6 +78,28 @@ public DLPreviewRenderer getThumbnailDLPreviewRenderer(
 		return null;
 	}
 
+	private void _checkForPreviewGenerationExceptions(FileVersion fileVersion)
+		throws PortalException {
+
+		if (_dlFileVersionPreviewLocalService.hasDLFileVersionPreview(
+				fileVersion.getFileEntryId(), fileVersion.getFileVersionId(),
+				DLFileVersionPreviewConstants.STATUS_FAILURE)) {
+
+			throw new DLFileEntryPreviewGenerationException();
+		}
+
+		if (!_videoProcessor.hasVideo(fileVersion)) {
+			if (!DLProcessorRegistryUtil.isPreviewableSize(fileVersion)) {
+				throw new DLPreviewSizeException();
+			}
+
+			throw new DLPreviewGenerationInProcessException();
+		}
+	}
```

Comment:
Can you name this method `_validateFileVersion`? In any case, validation methods should start with `_validate*`

Reply:
the name is the same on the audio (old code), thats why this name.

I will rename both

Reply:
Hi @adolfopa 

The name _checkForPreviewGenerationExceptions is used on all the _RendererProvider_ so I will not change it unless you want me to change it on all of them.

Reply:
If there's precedent, then it is fine. Keep it as is.